cli/azd/pkg/tools/kubectl/kube_config.go

```diff
@@ -0,0 +1,124 @@
+package kubectl
+
+import (
+	"context"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/osutil"
+	"gopkg.in/yaml.v3"
+)
+
+// Manages k8s configurations available to the k8s CLI
+type KubeConfigManager struct {
+	cli        KubectlCli
+	configPath string
+}
+
+// Creates a new instance of the KubeConfigManager
+func NewKubeConfigManager(cli KubectlCli) (*KubeConfigManager, error) {
+	kubeConfigDir, err := getKubeConfigDir()
+	if err != nil {
+		return nil, err
+	}
+
+	return &KubeConfigManager{
+		cli:        cli,
+		configPath: kubeConfigDir,
+	}, nil
+}
+
+// Parses the raw bytes into a KubeConfig instance
+func ParseKubeConfig(ctx context.Context, raw []byte) (*KubeConfig, error) {
+	var existing KubeConfig
+	if err := yaml.Unmarshal(raw, &existing); err != nil {
+		return nil, fmt.Errorf("failed unmarshalling Kube Config YAML: %w", err)
+	}
+
+	return &existing, nil
+}
+
+// Saves the KubeConfig to the kube configuration folder with the specified name
+func (kcm *KubeConfigManager) SaveKubeConfig(ctx context.Context, configName string, config *KubeConfig) error {
```

Comment:
Is this `kube config` something in the universe of `azd` ?  or are we using a file that is also used by aks?
If it is `azd specific`, should it be in `~.azd/` instead of `~/.kube` ?

Reply:
This is the `kubectl` local configuration.